NavigationReactNative/src/android/app/src/main/java/com/navigation/reactnative/NavigationBarView.java

```diff
@@ -0,0 +1,240 @@
+
+import android.content.Context;
+import android.content.res.Resources;
+import android.content.res.TypedArray;
+import android.graphics.Color;
+import android.graphics.drawable.Drawable;
+import android.os.Build;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.ViewOutlineProvider;
+
+import com.facebook.drawee.drawable.ScalingUtils;
+import com.facebook.drawee.generic.GenericDraweeHierarchy;
+import com.facebook.drawee.generic.GenericDraweeHierarchyBuilder;
+import com.facebook.drawee.view.DraweeHolder;
+import com.facebook.drawee.view.MultiDraweeHolder;
+import com.facebook.react.bridge.Arguments;
+import com.facebook.react.bridge.ReactContext;
+import com.facebook.react.bridge.ReadableArray;
+import com.facebook.react.bridge.ReadableMap;
+import com.facebook.react.bridge.WritableMap;
+import com.facebook.react.uimanager.events.RCTEventEmitter;
+import com.google.android.material.appbar.AppBarLayout;
+
+import androidx.annotation.Nullable;
+import androidx.appcompat.widget.Toolbar;
+
+public class NavigationBarView extends AppBarLayout {
+    private IconResolver iconResolver;
+
+    Toolbar toolbar;
+
+    int defaultContentInsetStart;
+    int defaultContentInsetEnd;
+    int defaultTitleTextColor;
+    ViewOutlineProvider defaultOutlineProvider;
+    Drawable defaultBackground;
+
+    private final DraweeHolder mLogoHolder = DraweeHolder.create(createDraweeHierarchy(), getContext());
+    private final DraweeHolder mNavIconHolder = DraweeHolder.create(createDraweeHierarchy(), getContext());
+    private final DraweeHolder mOverflowIconHolder = DraweeHolder.create(createDraweeHierarchy(), getContext());
+
+    private IconResolver.IconControllerListener mLogoControllerListener;
+    private IconResolver.IconControllerListener mNavIconControllerListener;
+    private IconResolver.IconControllerListener mOverflowIconControllerListener;
+    private final MultiDraweeHolder<GenericDraweeHierarchy> mActionsHolder =
+            new MultiDraweeHolder<>();
+
+    public NavigationBarView(Context context) {
+        super(context);
+
+        toolbar = new Toolbar(context);
+        addView(toolbar, ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT);
+
+        defaultContentInsetStart = toolbar.getContentInsetStart();
+        defaultContentInsetEnd = toolbar.getContentInsetEnd();
+        defaultTitleTextColor = getDefaultTitleTextColor(context);
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            defaultOutlineProvider = getOutlineProvider();
+        }
+        defaultBackground = getBackground();
+
+        iconResolver = new IconResolver(context);
+        mLogoControllerListener = iconResolver.new IconControllerListener(mLogoHolder) {
+            @Override
+            protected void setDrawable(Drawable d) {
+                toolbar.setLogo(d);
+                requestLayout();
```

Comment:
The calls to `requestLayout` don't match with the original React Native Toolbar code. The setDrawable of the logp, navigation and overflow icons [don't call requestLayout in the original](https://github.com/facebook/react-native/blob/56751851dfefd4b99d7cf478e6d04d9dc1aa4d11/ReactAndroid/src/main/java/com/facebook/react/views/toolbar/ReactToolbar.java#L144).  But the setDrawable of the menu item images do call [requestLayout in the original](https://github.com/facebook/react-native/blob/56751851dfefd4b99d7cf478e6d04d9dc1aa4d11/ReactAndroid/src/main/java/com/facebook/react/views/toolbar/ReactToolbar.java#L100). In our code this is reversed. The logo calls requestLayout but the menu items don't. Any reason for this change?